controllers/deployment.go

```diff
@@ -185,12 +197,22 @@ func buildStatefulSets(master *v1alpha1.CDAPMaster, name string, services Servic
 		}
 		env := addJavaMaxHeapEnvIfNotPresent(ss.Env, ss.Resources)
 		c := newContainerSpec(master, s, dataDir).setResources(ss.Resources).setEnv(env)
+		// Only main container should run a jmx server if enabed
+		if !isSidecar && metricsSidecarEnabled {
```

Comment:
Add sidecar outside of this loop if metrics collection is enabled. That way the additional check for sidecar can be avoided.

Reply:
Where is the SystemMetricExporterServiceSpec being used ?

Reply:
Added inside loop when service has system metrics enabled. I think since users can control which services to enable metrics for, they can use CR to ensure only one service should have this enabled per pod. Presently we have one service per pod, so such a situation should not arise. For adding container outside the loop, we'll also have code for this in 3 parts:

1. Before loop, initialising boolean to false
2. Inside loop, checking and setting to true for a service if required.
3. After loop, adding the container to sts spec.

So, i think having it in one place, inside the loop, seems better. Have simplified code according to review comments, so it should look cleaner now.